### PR TITLE
feat: track + notify user signups

### PR DESCRIPTION
**Pull request by sgfost**
> Enable participants to sign up for specific launch time(s) while viewing the launch schedule in their tournament / player dashboard.

- [x] provide a visual indicator of number of total signups in that time slot
- [x] create toggle button UI to "signup" or "un-signup" for a specific TournamentRoundDate
- [x] build backend endpoints to do the data things on the backend to signup / unsignup
- [x] build backend schedule logic to send an email reminder to eligible signed up participants when the lobby opens (and a few minutes before the lobby closes?)
- [x] automated tests for backend logic
- [x] prevent players that have already participated from signing up
- [x] de-activate signups when a player has participated
- [x] come up with a better indicator for the amt of players signed up (configurable threshold)

resolves virtualcommons/planning#63
resolves virtualcommons/planning#66

### CODE REVIEW

**Comment by sgfost** [comment]
Added an 'interest level' bar which should roughly show how many people are signed up for any given time-slot. Do you think its clear enough what's going on here @sabrinanel3?

Screenshot from 2024-01-24 15-22-47

File: client/src/api/tournament/request.ts [hunk 1]
```diff
@@ -25,4 +39,30 @@ export class TournamentAPI {
       throw e;
     }
   }
+
+  async addSignup(tournamentRoundDateId: number, inviteId: number): Promise<void> {
```
Comment (alee) [inline client/src/api/tournament/request.ts:43]: could merge these into a single function that adds a parameter `action=add|remove` that then posts to  `/tournament/signup/${action}`

Comment (sgfost) [reply client/src/api/tournament/request.ts:43]: Good call

**Review by alee — APPROVED** [review]
LGTM, good work @sgfost! Will do a bit of minor refactoring and then merge.

**Pull request merged**